library/runtime-api/src/commonMain/kotlin/io/matthewnelson/kmp/tor/runtime/api/config/builders/TCPPortBuilder.kt

```diff
@@ -0,0 +1,185 @@
+            this.port = port.toString()
+            return this
+        }
+
+        internal companion object {
+
+            @JvmSynthetic
+            internal fun build(
+                block: ThisBlock<Control>
+            ): String {
+                val b = Control().apply(block)
+
+                return b.port
+            }
+        }
+    }
+
+    /**
+     * For configuring [TorConfig.__SocksPort] to use a TCP Port.
+     * */
+    @KmpTorDsl
+    public class Socks private constructor(): TCPPortBuilder(),
+        DSLAuto<Socks>,
+        DSLDisable<Socks>,
+        DSLPort<Socks>
+    {
+        @get:JvmName("port")
+        public var port: String = "9050"
+            private set
+
+        @KmpTorDsl
+        public override fun auto(): Socks {
+            port = TorConfig.AUTO
+            return this
+        }
+
+        @KmpTorDsl
+        public override fun disable(): Socks {
+            port = "0"
+            return this
+        }
+
+        @KmpTorDsl
+        public override fun port(port: Port.Proxy): Socks {
+            this.port = port.toString()
+            return this
+        }
+
+        internal companion object {
+
+            @JvmSynthetic
+            internal fun build(
+                block: ThisBlock<Socks>
+            ): String {
+                val b = Socks().apply(block)
+
+                return b.port
+            }
+        }
+    }
+
+    /**
+     * For configuring [TorConfig.HiddenServicePort] to use a specific
+     * target TCP Port, instead of utilizing the specified
+     * [TorConfig.HiddenServicePort.virtual] port.
+     * */
+    @KmpTorDsl
+    public class HiddenService private constructor(): TCPPortBuilder() {
+
+        @JvmField
+        public var target: Port? = null
+
+        internal companion object {
+
+            @JvmSynthetic
+            internal fun build(
+                block: ThisBlock<HiddenService>
+            ): String? {
+                val b = HiddenService().apply(block)
+
+                return b.target?.toString()
+            }
+        }
+    }
+
+    @InternalKmpTorApi
+    public interface DSL<out T: TCPPortBuilder, out R: Any> {
+
+        /**
+         * For a [TorConfig.Keyword] that can be configured to use
+         * a TCP Port, or a Unix Socket.
+         * */
+        @KmpTorDsl
+        public fun asPort(
+            block: ThisBlock<T>,
+        ): R
+    }
+
+    @InternalKmpTorApi
+    public interface DSLAuto<out R: Any> {
+
+        /**
+         * Sets the port to "auto", indicating that Tor should
+         * pick an available port.
+         * */
+        @KmpTorDsl
+        public fun auto(): R
+    }
+
+    @InternalKmpTorApi
+    public interface DSLDisable<out R: Any> {
+
+        /**
+         * Disables the [TorConfig.Keyword] by setting its port
+         * to "0".
+         * */
+        @KmpTorDsl
+        public fun disable(): R
+    }
+
+    // TODO: IPAddress/Localhost
```

Comment:
Need to add ability to configure an IP address

Reply:
Created #313 for this capability. It is pretty involved and needs some more things built out before implementing.